src/stores/ViewportStore.ts

```diff
@@ -0,0 +1,131 @@
+import Store from '@/stores/Store'
+import { Action } from '@/stores/Dispatcher'
+import {
+    InfoReceived,
+    PathDetailsRangeSelected,
+    RouteRequestSuccess,
+    SetSelectedPath,
+    SetViewport,
+    SetViewportToPoint,
+} from '@/actions/Actions'
+import { FlyToInterpolator, WebMercatorViewport } from 'react-map-gl'
+import TransitionInterpolator from 'react-map-gl/src/utils/transition/transition-interpolator'
+import { Bbox } from '@/api/graphhopper'
+import RouteStore from '@/stores/RouteStore'
+
+export interface ViewportStoreState {
+    longitude: number
+    latitude: number
+    zoom: number
+    width: number
+    height: number
+    altitude?: number
+    bearing?: number
+    minZoom?: number
+    maxZoom?: number
+    minPitch?: number
+    maxPitch?: number
+    transitionDuration?: number
+    transitionEasing?: Function
+    transitionInterpolator?: TransitionInterpolator
+    transitionInterruption?: number
+}
+
+// have this right here for now. Not sure if this needs to be abstracted somewhere else
+const mediaQuery = window.matchMedia('(max-width: 640px)')
```

Comment:
This makes testing the store a bit complicated, see my other comments. Can we just inject this into the constructor? And is it the same as `isSmallScreen` in `App.tsx`? Here we use `640px` and in `App.tsx` we have something similar with `44rem`.

Reply:
Yes, it would be better to have this media query at 44rem as well. And yes, it is supposed to be the same as ```isSmallScreen```.

I guess you could even inject something like ```isSmallScreen: () => boolean``` and pass ```new ViewportStore(() => window.matchMedia(...))``` in ```index.tsx```. Then in the test you could supply your own predicate.

Reply:
I did this. The only thing is that we have this media query twice now. One time in `index.tsx` where we use `window.matchMedia` and one time in `App.tsx` where we use the `useMediaQuery` hook. We cannot use the hook in `index.tsx` because it would be outside of any react component and I thought there is probably a reason you use the `useMediaQuery` hook in `App.tsx`

Reply:
Yes, the hook is used, so that the layout changes when the window is resized. 

There are several other places in css where this media query is used as well. I think as long as we only have one breakpoint we may be fine.